dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -165,4 +196,46 @@ public CommunityResource wrapResource(CommunityRest community, String... rels) {
         return new CommunityResource(community, utils, rels);
     }
 
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected CommunityRest put(Context context, HttpServletRequest request, String apiCategory, String model, UUID id,
+                       JsonNode jsonNode)
+        throws RepositoryMethodNotImplementedException, SQLException, AuthorizeException {
+        CommunityRest communityRest = new Gson().fromJson(jsonNode.toString(), CommunityRest.class);
+        Community community = cs.find(context, id);
+        if (community == null) {
+            throw new ResourceNotFoundException(apiCategory + "." + model + " with id: " + id + " not found");
+        }
+        CommunityRest originalCommunityRest = converter.fromModel(community);
+        if (communityRestEqualityUtils.isCommunityRestEqualWithoutMetadata(originalCommunityRest, communityRest)) {
+            List<MetadataEntryRest> metadataEntryRestList = communityRest.getMetadata();
+            community = (Community) dspaceObjectUtils.replaceMetadataValues(context, community, metadataEntryRestList);
+        } else {
+            throw new UnprocessableEntityException("The given JSON and the original Community differ more " +
+                                                       "than just the metadata");
+        }
+        return converter.fromModel(community);
+    }
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected void delete(Context context, UUID id) throws AuthorizeException {
```

Comment:
Should we check for DELETE on the community instead than general admin rights?

Reply:
@abollini is correct. this annotation could be updated to
`@PreAuthorize("hasPermission(#id, 'COMMUNITY', 'DELETE')")`